npm/vite-dev-server/src/index.ts

```diff
@@ -0,0 +1,31 @@
+import { EventEmitter } from 'events'
+import { debug as debugFn } from 'debug'
+import { start as createDevServer } from './startServer'
+import { UserConfig } from 'vite'
+const debug = debugFn('cypress:vite-dev-server:vite')
+
+interface Options {
+  specs: any[] // Cypress.Cypress['spec'][] // Why isn't this working? It works for webpack-dev-server
+  config: Record<string, string>
+  devServerEvents: EventEmitter
+  [key: string]: unknown
+}
+
+export interface StartDevServer {
+  /* this is the Cypress options object */
+  options: Options
+  /* support passing a path to the user's webpack config */
+  viteConfig?: UserConfig // TODO: implement taking in the user's vite configuration. Right now we don't
+}
+
+export async function startDevServer (startDevServerArgs: StartDevServer): Promise<number> {
+  const viteDevServer = await createDevServer(startDevServerArgs)
+
+  return new Promise(async (resolve) => {
+    const app = await viteDevServer.listen()
+    const port = app.config.server.port
+
+    debug('Component testing vite server started on port', port)
+    resolve(port)
```

Comment:
I changed this to resolve an object of options a few days ago, to make writing tests for the webpack-dev-server easier. Can you update this to match that API: https://github.com/cypress-io/cypress/pull/14479/commits/81b9ad3b2a78f3bde7253a61e60639c8649e8eba#diff-e64b391b74ff6abced0531e5b572ea473550755d48eb5cda19bda25c7192d0b0R36

Reply:
Not really. We don't really get the raw httpSrvr. We get a Connect server.

Reply:
I don't understand you comment - what do you mean by "not really".

At the moment we resolve a port - to be able to close the server at the end of each test in the `webpack-dev-server` tests, I needed to get access to the actual server, so I changed this to resolve the server and port in an object instead of just the port.

As this stands, once you merge in the latest `feat/component-testing`, this will not work, since we now expect to resolve an object here, not just a port. [See here](https://github.com/cypress-io/cypress/blob/b0f324b2ee98714e882551d78c89283fbe3f9cc0/packages/server-ct/src/project-ct.ts#L107).

We can change this API if you disagree with it, but I don't see any way to access the undying server in the tests. [See these tests for an example](https://github.com/cypress-io/cypress/blob/b0f324b2ee98714e882551d78c89283fbe3f9cc0/npm/webpack-dev-server/test/e2e.spec.ts#L58). I needed to be able to call `server.close()` at the end of each one.

Reply:
I meant "not really" because I just didn't find the httpServer on the API. After digging through the source code, I found it [here](https://github.com/vitejs/vite/blob/dbe1f4aeadfb9f927f8cd0eec95a32aeee9c3601/packages/vite/src/node/server/index.ts#L324). I'll update. In the future we should probably standardize the API and not rely on getting a raw httpServer and instead provide a close method whose signature is standardized.